janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/VertexProgramScanJob.java

```diff
@@ -116,20 +120,21 @@ public void process(JanusGraphVertex vertex, ScanMetrics metrics) {
 
     @Override
     public void getQueries(QueryContainer queries) {
-        if (vertexProgram instanceof TraversalVertexProgram) {
+        Set<MessageScope> previousScopes = vertexMemory.getPreviousScopes();
+        if (vertexProgram instanceof TraversalVertexProgram || vertexProgram instanceof ShortestPathVertexProgram ||
+            vertexProgram instanceof ConnectedComponentVertexProgram || previousScopes.contains(globalScope)) {
             //TraversalVertexProgram currently makes the assumption that the entire star-graph around a vertex
             //is available (in-memory). Hence, this special treatment here.
             //TODO: After TraversalVertexProgram is adjusted, remove this
+            //Special handling for ShortestPathVertexProgram necessary until TINKERPOP-2187 is resolved
+            //Apparently, ConnectedComponentVertexProgram also needs this special handling
             queries.addQuery().direction(Direction.BOTH).edges();
-            return;
         }
 
-        for (MessageScope scope : vertexMemory.getPreviousScopes()) {
-            if (scope instanceof MessageScope.Global) {
-                queries.addQuery().direction(Direction.BOTH).edges();
```

Comment:
What is happened with this?

Reply:
I included this case in the `if` clause above as that now contains this condition: `previousScopes.contains(globalScope)` with `Set<MessageScope> previousScopes = vertexMemory.getPreviousScopes();` and then it does exactly the same thing as before:
```java
queries.addQuery().direction(Direction.BOTH).edges();
```

I needed the same handling for the other cases in the `if` clause above, so I joined them with this case of `MessageScope.Global`.